torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -231,13 +230,7 @@ def _init_device_mesh(
 
     device_type = root_state.compute_device.type
     mesh_tensor = torch.arange(get_world_size(root_state.process_group))
-    device_mesh = DeviceMesh(device_type, mesh_tensor, _init_process_groups=False)
-    device_mesh._dim_group_infos = [
-        (
-            _get_group_tag(root_state.process_group),
-            mesh_tensor.tolist(),
-        )
-    ]
+    device_mesh = DeviceMesh(device_type, mesh_tensor, _validate_mesh=False)
```

Comment:
This would still try to calls into the `init_process_groups` function inside device mesh and only works for the case where the mesh_tensor is the same as world_pg, otherwise it would still eventually calls `new_group`. I am wondering if we should put the logic of `skip` pg creation we discussed together in this PR, or you can make a follow up PR to skip pg creation if necessary, then delete `_init_process_groups` flag